File: cognix-flask-server/test_clinical_api.py
"""
Test Clinical Assessment API
Tests the new clinical screening endpoints
"""

import requests
import json

BASE_URL = "http://localhost:5000"

def test_analyze_answers():
    """Test the /clinical/analyze endpoint"""
    print("\n" + "="*60)
    print("TEST 1: Analyzing Text Answers")
    print("="*60)
    
    # Test cognitive domain
    payload = {
        "domain": "cognitive",
        "answers": [
            "Yes, I frequently forget recent conversations and where I put things",
            "I find it very difficult to plan my day or solve simple problems",
            "Sometimes I get confused about what day it is",
            "Daily tasks like cooking are much harder than before"
        ]
    }
    
    response = requests.post(f"{BASE_URL}/clinical/analyze", json=payload)
    print(f"\nStatus Code: {response.status_code}")
    print(f"Response: {json.dumps(response.json(), indent=2)}")
    
    return response.json()


def test_clinical_prediction():
    """Test the /clinical/predict endpoint"""
    print("\n" + "="*60)
    print("TEST 2: Clinical Prediction with 4 Domain Scores")
    print("="*60)
    
    # Example: Patient with moderate cognitive and motor symptoms
    payload = {
        "demographic_score": 3.5,
        "cognitive_score": 7.2,
        "motoric_score": 4.8,
        "neuropsychiatric_score": 2.1,
        "user_answers": {
            "demographic": [
                "I am 68 years old",
                "Sometimes I get confused about my exact age",
                "My family has corrected me a few times",
                "Age-related details sometimes confuse me"
            ],
            "cognitive": [
                "Yes, I frequently forget recent conversations",
                "I find it very difficult to plan or solve daily problems",
                "I often feel confused about time and date",
                "Familiar tasks are much harder than before"
            ],
            "motoric": [
                "I have noticed slight shaking in my hands",
                "My muscles feel a bit stiff sometimes",
                "I feel slower while walking",
                "I have some balance problems"
            ],
            "neuropsychiatric": [
                "No, I haven't seen things that aren't there",
                "No unusual sounds or voices",
                "I sleep normally",
                "My alertness is fine"
            ]
        }
    }
    
    response = requests.post(f"{BASE_URL}/clinical/predict", json=payload)
    print(f"\nStatus Code: {response.status_code}")
    print(f"Response: {json.dumps(response.json(), indent=2)}")
    
    return response.json()


def test_full_workflow():
    """Test complete workflow: analyze answers -> get scores -> predict"""
    print("\n" + "="*60)
    print("TEST 3: Full Clinical Assessment Workflow")
    print("="*60)
    
    # Step 1: Define all answers
    all_answers = {
        "demographic": [
            "I am 72 years old",
            "Yes, I sometimes feel confused about my age",
            "My family has corrected me multiple times",
            "Age-related details often confuse me"
        ],
        "cognitive": [
            "Yes, I always forget recent conversations and events",
            "I find it extremely difficult to plan or solve problems",
            "I am constantly confused about time, date, and place",
            "Daily tasks are very hard now"
        ],
        "motoric": [
            "Yes, I have severe shaking in my hands and legs",
            "My muscles are very stiff and rigid",
            "I am extremely slow while walking",
            "I frequently have balance problems and fall"
        ],
        "neuropsychiatric": [
            "Yes, I often see things that others say aren't there",
            "I frequently hear voices without a source",
            "I constantly move and talk during sleep",
            "I have severe changes in alertness"
        ]
    }
    
    # Step 2: Analyze each domain
    scores = {}
    for domain, answers in all_answers.items():
        print(f"\n📊 Analyzing {domain.upper()} domain...")
        response = requests.post(
            f"{BASE_URL}/clinical/analyze",
            json={"domain": domain, "answers": answers}
        )
        if response.status_code == 200:
            result = response.json()
            scores[f"{domain}_score"] = result['severity_score']
            print(f"   ✅ Score: {result['severity_score']}/10")
        else:
            print(f"   ❌ Error: {response.text}")
            return
    
    # Step 3: Make prediction
    print(f"\n🧠 Making prediction with scores: {scores}")
    scores['user_answers'] = all_answers
    
    response = requests.post(f"{BASE_URL}/clinical/predict", json=scores)
    
    if response.status_code == 200:
        result = response.json()
        prediction = result['prediction']
        
        print("\n" + "="*60)
        print("🎯 PREDICTION RESULTS")
        print("="*60)
        print(f"\n🏥 Predicted Disease: {prediction['predicted_disease']}")
        print(f"📊 Confidence: {prediction['confidence']*100:.1f}%")
        print(f"\n📈 All Probabilities:")
        for disease, prob in prediction['probabilities'].items():
            print(f"   {disease:20s}: {prob*100:5.1f}%")
        print("\n" + "="*60)
    else:
        print(f"❌ Prediction Error: {response.text}")


if __name__ == "__main__":
    print("\n🧪 CLINICAL ASSESSMENT API TESTS")
    print("="*60)
    
    try:
        # Test 1: Analyze answers
        test_analyze_answers()
        
        # Test 2: Direct prediction with scores
        test_clinical_prediction()
        
        # Test 3: Full workflow
        test_full_workflow()
        
        print("\n✅ All tests completed!")
        
    except requests.exceptions.ConnectionError:
        print("\n❌ ERROR: Cannot connect to Flask server")
        print("   Make sure the server is running on http://localhost:5000")
    except Exception as e:
        print(f"\n❌ ERROR: {str(e)}")

File: cognix-server/test-clinical-api.js
/**
 * Test Clinical Assessment Node.js API
 * Tests the integration between Node.js and Flask for clinical assessments
 */

const axios = require('axios');

const NODE_API_URL = 'http://localhost:3000';
const TEST_USER_ID = 'test_user_' + Date.now();

async function testClinicalAssessment() {
  console.log('\n' + '='.repeat(60));
  console.log('🧪 TESTING CLINICAL ASSESSMENT FLOW');
  console.log('='.repeat(60));

  try {
    // Step 1: Define all user answers
    const allAnswers = {
      demographic: [
        'I am 72 years old',
        'Yes, I sometimes feel confused about my age',
        'My family has corrected me a few times',
        'Age-related details sometimes confuse me'
      ],
      cognitive: [
        'Yes, I frequently forget recent conversations',
        'I find it very difficult to plan my day',
        'I often get confused about the date',
        'Daily tasks are harder than before'
      ],
      motoric: [
        'I have noticed some shaking in my hands',
        'My muscles feel stiff sometimes',
        'I feel slower while walking',
        'I have occasional balance problems'
      ],
      neuropsychiatric: [
        'No, I haven\'t seen things that aren\'t there',
        'No unusual sounds',
        'I sleep normally',
        'My alertness is fine'
      ]
    };

    // Step 2: Analyze each domain and collect scores
    console.log('\n📊 STEP 1: Analyzing Answers...\n');
    const scores = {};

    for (const [domain, answers] of Object.entries(allAnswers)) {
      console.log(`   Analyzing ${domain.toUpperCase()} domain...`);
      
      const response = await axios.post(`${NODE_API_URL}/api/clinical/analyze`, {
        domain,
        answers
      });

      if (response.data.success) {
        scores[`${domain}_score`] = response.data.severity_score;
        console.log(`   ✅ Score: ${response.data.severity_score}/10`);
      } else {
        console.log(`   ❌ Error: ${response.data.error}`);
        return;
      }
    }

    // Step 3: Complete assessment and save to database
    console.log('\n🧠 STEP 2: Completing Assessment...\n');
    
    const assessmentPayload = {
      userId: TEST_USER_ID,
      ...scores,
      user_answers: allAnswers
    };

    const assessmentResponse = await axios.post(
      `${NODE_API_URL}/api/clinical/assess`,
      assessmentPayload
    );

    if (assessmentResponse.data.success) {
      const result = assessmentResponse.data.data;
      
      console.log('   ✅ Assessment Saved Successfully!');
      console.log(`   📝 History ID: ${result.historyId}`);
      
      console.log('\n' + '='.repeat(60));
      console.log('🎯 PREDICTION RESULTS');
      console.log('='.repeat(60));
      console.log(`\n🏥 Predicted Disease: ${result.prediction.predicted_disease}`);
      console.log(`📊 Confidence: ${(result.prediction.confidence * 100).toFixed(1)}%`);
      console.log(`⚠️  Risk Level: ${result.prediction.riskLevel}`);
      
      console.log('\n📈 All Probabilities:');
      for (const [disease, prob] of Object.entries(result.prediction.probabilities)) {
        console.log(`   ${disease.padEnd(20)}: ${(prob * 100).toFixed(1)}%`);
      }
      
      console.log('\n📊 Domain Scores:');
      for (const [domain, score] of Object.entries(result.domainScores)) {
        console.log(`   ${domain.padEnd(20)}: ${score}/10`);
      }
      
      console.log('\n' + '='.repeat(60));

      // Step 4: Retrieve assessment history
      console.log('\n📚 STEP 3: Retrieving Assessment History...\n');
      
      const historyResponse = await axios.get(
        `${NODE_API_URL}/api/clinical/history/${TEST_USER_ID}`
      );

      if (historyResponse.data.success) {
        console.log(`   ✅ Found ${historyResponse.data.data.length} assessment(s)`);
        console.log(`   📊 Total: ${historyResponse.data.pagination.total}`);
      }

      // Step 5: Get specific assessment details
      console.log('\n🔍 STEP 4: Retrieving Assessment Details...\n');
      
      const detailsResponse = await axios.get(
        `${NODE_API_URL}/api/clinical/assessment/${result.historyId}`
      );

      if (detailsResponse.data.success) {
        const assessment = detailsResponse.data.data;
        console.log('   ✅ Assessment Details Retrieved');
        console.log(`   📅 Date: ${new Date(assessment.createdAt).toLocaleString()}`);
        console.log(`   🧠 Disease: ${assessment.prediction.disease}`);
        console.log(`   📊 Confidence: ${(assessment.prediction.confidence * 100).toFixed(1)}%`);
      }

      console.log('\n' + '='.repeat(60));
      console.log('✅ ALL TESTS PASSED!');
      console.log('='.repeat(60) + '\n');

    } else {
      console.log(`   ❌ Assessment Error: ${assessmentResponse.data.error}`);
    }

  } catch (error) {
    console.error('\n❌ ERROR:', error.response?.data || error.message);
    if (error.code === 'ECONNREFUSED') {
      console.log('\n💡 Make sure both servers are running:');
      console.log('   - Flask server: http://localhost:5000');
      console.log('   - Node.js server: http://localhost:3000');
    }
  }
}

// Run the test
testClinicalAssessment();
